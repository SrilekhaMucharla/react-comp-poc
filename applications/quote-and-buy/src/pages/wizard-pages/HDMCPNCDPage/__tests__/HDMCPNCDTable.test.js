import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HDMCPNCDTable from '../HDMCPNCDTable';

Enzyme.configure({ adapter: new Adapter() });

describe('<HDPNCDTable />', () => {
    let wrapper = null;

    beforeEach(async () => {
        wrapper = await mount(
            <HDMCPNCDTable
                ncdData={[
                    { years: '2', discount: 15 },
                    { years: '3', discount: 25 },
                    { years: '4', discount: 28 },
                    { years: '5', discount: 30 },
                ]} />,

        );
    });

    it('should render the component and match snapshot', async () => {
        expect(wrapper).toMatchSnapshot();
    });
});
