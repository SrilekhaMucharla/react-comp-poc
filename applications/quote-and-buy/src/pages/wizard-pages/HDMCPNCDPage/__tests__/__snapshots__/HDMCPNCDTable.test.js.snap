// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDPNCDTable /> should render the component and match snapshot 1`] = `
<HDPNCDTable
  ncdData={
    Array [
      Object {
        "discount": 15,
        "years": "2",
      },
      Object {
        "discount": 25,
        "years": "3",
      },
      Object {
        "discount": 28,
        "years": "4",
      },
      Object {
        "discount": 30,
        "years": "5",
      },
    ]
  }
>
  <h3>
    Step-back table
  </h3>
  <table
    className="step-back-table"
  >
    <thead
      className="step-back-table__head"
    >
      <tr
        className="step-back-table__row"
      >
        <th
          className="step-back-table__row__th"
        >
          Number of years NCD
        </th>
        <th
          className="step-back-table__row__th"
        >
          Average NCD discount last year
        </th>
      </tr>
    </thead>
    <tbody
      className="step-back-table__body"
    >
      <tr
        className="step-back-table__body-row"
      >
        <td
          className="step-back-table__body-row__td step-back-table__years"
        >
          2
           years
        </td>
        <td
          className="step-back-table__body-row__td step-back-table__discout"
        >
          15
          %
        </td>
      </tr>
      <tr
        className="step-back-table__body-row"
      >
        <td
          className="step-back-table__body-row__td step-back-table__years"
        >
          3
           years
        </td>
        <td
          className="step-back-table__body-row__td step-back-table__discout"
        >
          25
          %
        </td>
      </tr>
      <tr
        className="step-back-table__body-row"
      >
        <td
          className="step-back-table__body-row__td step-back-table__years"
        >
          4
           years
        </td>
        <td
          className="step-back-table__body-row__td step-back-table__discout"
        >
          28
          %
        </td>
      </tr>
      <tr
        className="step-back-table__body-row"
      >
        <td
          className="step-back-table__body-row__td step-back-table__years"
        >
          5
           years
        </td>
        <td
          className="step-back-table__body-row__td step-back-table__discout"
        >
          30
          %
        </td>
      </tr>
    </tbody>
  </table>
</HDPNCDTable>
`;
